src/DDTrace/ScopeManager.php

```diff
@@ -57,6 +76,63 @@ public function deactivate(Scope $scope)
         }
 
         array_splice($this->scopes, $i, 1);
+
+        $span = $scope->getSpan();
+        if ($span instanceof Span && isset($span->ddtrace_scope_activated)) {
```

Comment:
I think that `$span instanceof Span` should be removed. If the span has a property `$span->ddtrace_scope_activated` then it has to be kept in sync.

Reply:
Not quite. `Span` and only Span is guaranteed to be backed by an internal span. Everything else is assumed to be custom. (i.e. it's designed as an implementation detail of `Span`)

Reply:
I am resolving this comment as the code works. But I am not sold on this. I do not see any reasonable reason why an `<object>` having a property `->ddtrace_scope_activated` should not be kept in sync.